app/controllers/dialcode/DialcodeV3Controller.java

```diff
@@ -76,7 +76,16 @@ public Promise<Result> searchDialCode() {
         Request request = getRequest();
         try {
             Map<String, Object> map = (Map<String, Object>) request.get(DialCodeEnum.search.name());
-            Response response = dialCodeManager.searchDialCode(request.getContext(), map);
+            List<String> fieldsList = new ArrayList<String>();
+            if(request.get(DialCodeEnum.fields.name()) != null) {
+                try {
+                    fieldsList = (List<String>) request.get(DialCodeEnum.fields.name());
+                } catch (ClassCastException ce) {
+                    fieldsList.add(request.get(DialCodeEnum.fields.name()).toString());
+                }
+                fieldsList.add("identifier");
```

Comment:
This code doesn't work if we give `fields` as Map. It should not give 500 error. Please handle this also...